### PR TITLE
Move to dune for OCaml 5 compatibility

### PR DESCRIPTION
**Pull request by 2BitSalute**
OCaml 5 is removing deprecated APIs, which makes this library fail to compile. Moving to dune removes the dependencies on oasis, etc.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,49 +1,15 @@
...
-
-configure:
-	$(SETUP) -configure $(CONFIGUREFLAGS)
-
-.PHONY: build doc test all install uninstall reinstall clean distclean configure
-
-# OASIS_STOP
+	@dune clean
 
-VERSION=$(shell oasis query version)
```
Comment (rr0gi) [inline Makefile:43]: please keep release target

File: README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# OCaml LZ4 bindings [![build](https://github.com/ahrefs/ocaml-murmur3/actions/workflows/main.yml/badge.svg)](https://github.com/ahrefs/ocaml-murmur3/actions/workflows/main.yml)
```
Comment (rr0gi) [inline README.md]:
```suggestion
# OCaml MurmurHash bindings [![build](https://github.com/ahrefs/ocaml-murmur3/actions/workflows/main.yml/badge.svg)](https://github.com/ahrefs/ocaml-murmur3/actions/workflows/main.yml)
```

File: dune [hunk 3]
```diff
@@ -0,0 +1,3 @@
+(env
+ (_
+  (flags :standard -strict-sequence -safe-string -warn-error -a+8)))
```
Comment (rr0gi) [inline dune]: I would omit warn-error altogether, otherwise why this specific choice of warnings?

File: .github/workfows/main.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+jobs:
+  run:
+    name: build
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          #- windows-latest
+          #- macos-latest
+        ocaml-compiler:
```
Comment (rr0gi) [inline .github/workfows/main.yml]: is there a 5.0 compiler?

**Review by rr0gi — CHANGES_REQUESTED** [review]
thanks! some minor comments

File: .github/workfows/main.yml (hunk 4 above)
Comment (2BitSalute) [reply .github/workfows/main.yml]: It's not released yet. The latest version is [5.0.0~alpha0](https://discuss.ocaml.org/t/ocaml-5-0-zeroth-alpha-release/10026), available as of June.

File: Makefile (hunk 1 above)
Comment (2BitSalute) [reply Makefile:43]: I'm trying to find a good method of doing the same thing with Dune. As far as I can tell, there isn't a built-in way like `oasis query`

File: README.md (hunk 2 above)
Comment (2BitSalute) [reply README.md]: Oops

File: Makefile (hunk 1 above)
Comment (rr0gi) [reply Makefile:43]: I am ok with populating VERSION manually in Makefile

File: dune (hunk 3 above)
Comment (2BitSalute) [reply dune]: This is to turn all warnings except for 8 (missing cases in pattern-matching) into non-fatal warnings to make development less obtrusive locally. I don't think it's important to keep this here. I'll remove.

Comment (2BitSalute) [inline dune]:
```suggestion
  (flags :standard -strict-sequence -safe-string)))
```

File: .github/workfows/main.yml (hunk 4 above)
Comment (rr0gi) [reply .github/workfows/main.yml]: there is alpha1 too.
I mean if it is available for github actions

File: .github/workfows/main.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+          - 4.03.x
+          - 4.13.x
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v2
+    - uses: ocaml/setup-ocaml@v2
+      with:
+        ocaml-compiler: ${{ matrix.ocaml-compiler }}
+    - run: opam pin -n .
+    - run: opam depext -yt murmur3
```
Comment (2BitSalute) [inline .github/workfows/main.yml:26]: @rr0gi I changed the name of the package to `murmur3` from `ocaml-murmur3` - would you rather keep the name?

File: Makefile (hunk 1 above)
Comment (2BitSalute) [reply Makefile:43]: Actually, I can make an exe that outputs the version using `dune-build-info` library. I think it's going to do the same thing as with `oasis`.

E.g., I see this as the output of `make release` after I make the change:
```
git tag -a -m "release 0.2.1" v0.2.1
git archive --prefix=murmur3-0.2.1/ v0.2.1 | gzip > murmur3-0.2.1.tar.gz
```

Does this look right?

File: .github/workfows/main.yml (hunk 4 above)
Comment (2BitSalute) [reply .github/workfows/main.yml]: OK, I found (was told) the right name. I'll update the PR in a bit.
I also found out why my main.yml wasn't triggering actions - I had a typo in the `workflow` folder name

File: Makefile [hunk 6]
```diff
@@ -1,49 +1,22 @@
...
-build: setup.data
-	$(SETUP) -build $(BUILDFLAGS)
-
-doc: setup.data build
-	$(SETUP) -doc $(DOCFLAGS)
-
-test: setup.data build
-	$(SETUP) -test $(TESTFLAGS)
+VERSION=$(shell dune exec src/version.exe)
+NAME=murmur3-$(VERSION)
```
Comment (2BitSalute) [inline Makefile]: @rr0gi,  I changed the name from `ocaml-murmur3` to `murmur3` - is that alright?

**Comment by 2BitSalute** [comment]
OK, I think I've now resolved all the issues, except that I'm having second thoughts about the package name change from `ocaml-murmur3` to `murmur3`. Maybe I should just keep it as it was.

File: Makefile (hunk 6 above)
Comment (2BitSalute) [reply Makefile]: I decided to change it back to ocaml-murmur3.

**Comment by 2BitSalute** [comment]
> OK, I think I've now resolved all the issues, except that I'm having second thoughts about the package name change from `ocaml-murmur3` to `murmur3`. Maybe I should just keep it as it was.

Decided to keep it as is.
In my fork, on the latest commit, the action results are successful:
https://github.com/2BitSalute/ocaml-murmur3/pull/5/checks

Includes builds on `4.08.1`, `4.13.1`, and `5.0.0~alpha1`.

File: Makefile (hunk 6 above)
Comment (rr0gi) [reply Makefile]: better not change to not cause breakage, yes

Comment (2BitSalute) [reply Makefile]: Yep, already did (latest version should be correct)

**Comment by 2BitSalute** [comment]
> thanks! some minor comments

I've addressed all the requested changes. Let me know if there's anything else I should do.

**Comment by rr0gi** [comment]
squashed and pushed to master aaa77351c31fb3ef629c5977d752cee880519ae6
thanks! (also fixed the name to murmur3 as it was properly and dropped dune-build-info dep)

**Pull request closed without merging**